### PR TITLE
Dispatch Page add a "End The Exam" button

### PR DESCRIPTION
**Pull request by adinKent**
[https://trello.com/c/CHqXUTDI] Dispatch Page add a "End The Exam" button
feature: implementat feature

### CODE REVIEW

File: src/redux/record/actions.js [hunk 1]
```diff
@@ -83,6 +83,40 @@ function updateRecordData(id, newCode) {
...
+function endRecordData(id) {
+  return async dispatch => {
+    dispatch(
+      graphqlActionHelper({
+        method: 'UPDATE',
+        dataName: 'RECORD',
+        actionState: ACTION_STATE.STARTED,
+      }),
+    );
+    try {
```
Comment (hank7444) [inline src/redux/record/actions.js:95]: 如果 endRecord(id), id 不存在, graphql 不會跳到 catch 區塊吧?

File: src/containers/DispatchPage/ControlWidget/index.js [hunk 2]
```diff
@@ -68,29 +77,48 @@ const ControlWidget = ({
...
               onChange={onChangeCategory}
               categoryIndex={categoryIndex}
             />
             <QuestionSelector
+              disabled={recordStatus === RECORD_STATUS.inprogress}
               onChange={onChangeQuestion}
               questionIndex={questionIndex}
               list={questionList}
             />
+            {recordStatus === RECORD_STATUS.inprogress || recordStatus === RECORD_STATUS.closed ?
```
Comment (hank7444) [inline src/containers/DispatchPage/ControlWidget/index.js]: 如果把三元運算子拆開會不會比較好閱讀?
```
{recordStatus === RECORD_STATUS.inprogress || recordStatus === RECORD_STATUS.closed &&
  condition 1
}

{recordStatus !== RECORD_STATUS.inprogress && recordStatus !== RECORD_STATUS.closed &&
 condition 2
}
```

File: src/redux/record/actions.js (hunk 1 above)
Comment (adinKent) [reply src/redux/record/actions.js:95]: I have tried to use an invalid id. It jumped to catch block.

File: src/containers/DispatchPage/ControlWidget/index.js (hunk 2 above)
Comment (adinKent) [reply src/containers/DispatchPage/ControlWidget/index.js]: Done.

Comment (pohsiangchen) [reply src/containers/DispatchPage/ControlWidget/index.js]: Hey Kent, You can find record status from currentRecord in Room model. Is that what you want?
image

File: src/redux/record/actions.js (hunk 1 above)
Comment (hank7444) [reply src/redux/record/actions.js:95]: why I tried to delete a room, it wouldn't jump to catch block? So weired.

Comment (adinKent) [reply src/redux/record/actions.js:95]: Maybe the error handling of operation update is different!?

**Pull request merged**